src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -206,11 +206,28 @@ private async Task ExecuteAsync(HttpContext context, ConnectionDelegate connecti
                 case HttpTransportType.None:
                     break;
                 case HttpTransportType.WebSockets:
+                    var isReconnect = connection.ApplicationTask is not null;
                     var ws = new WebSocketsServerTransport(options.WebSockets, connection.Application, connection, _loggerFactory);
                     if (!connection.TryActivatePersistentConnection(connectionDelegate, ws, currentRequestTcs.Task, context, _logger))
                     {
                         return;
                     }
+
+                    if (connection.UseAcks && isReconnect)
+                    {
+                        // Should call this after the transport has started, otherwise we'll be writing to a Pipe that isn't being read from
+                        try
+                        {
+                            var reconnectTask = connection.NotifyOnReconnect?.Invoke(connection.Transport.Output) ?? Task.CompletedTask;
+                            await reconnectTask;
```

Comment:
Do we really want to await resending everything before disabling the request body timeout? It seems like that could cause us to close healthy connections. Is there any possible benefit to awaiting this task immediately rather than putting it in the WhenAny?

The same goes on the client side even though there's no IHttpRequestTimeoutFeature. Why would the implementor of NotifyOnReconnect ever want to delay execution of anything at the Http.Connections layer? What's the `reconnectTask` for? @davidfowl

Reply:
> Is there any possible benefit to awaiting this task immediately rather than putting it in the WhenAny?

Well the `WhenAny` is for when the connection is closing, either due to the transport detecting a disconnect or the application completing. So we'd still want to wait for the reconnect task separately. But we should put that after the IHttpRequestTimeoutFeature.